### PR TITLE
api/distributedcontext: add Map unit tests

### PR DESCRIPTION
**Pull request by clsung**

### CODE REVIEW

File: api/distributedcontext/map_test.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+			wantKVs: []core.KeyValue{
+				key.New("key1").String("val1"),
+				key.New("key5").Int(5),
+			},
+		},
+		{
+			name: "Map with MapUpdate",
+			value: MapUpdate{SingleKV: key.New("key1").Int64(3),
+				MultiKV: []core.KeyValue{
+					key.New("key1").String(""),
```
Comment (jmacd) [inline api/distributedcontext/map_test.go]: nit: We have recently added syntax like `key.String("key1", "")` to simplify these expressions. Would you consider using this syntax?

Comment (clsung) [reply api/distributedcontext/map_test.go]: Sure, no problem.

File: api/distributedcontext/map_test.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+	"go.opentelemetry.io/api/core"
+	"go.opentelemetry.io/api/key"
+)
+
+func TestMap(t *testing.T) {
+	for _, testcase := range []struct {
+		name    string
+		init    []int
+		value   MapUpdate
```
[the remark this answers is not in the document]
Comment (clsung) [reply api/distributedcontext/map_test.go]: Updated, thanks.

**Comment by rghetia** [comment]
@clsung can you please rebase this? build is failing because everything moved under otel dir after #274.

**Comment by clsung** [comment]
Rebased, thanks.

**Pull request merged**